product_end_of_life/models/ir_cron.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# Integrators License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class IrCron(models.Model):
+    _inherit = 'ir.cron'
+
+    approaching_number = fields.Integer(
```

Comment:
:thinking: isn't this a global settings for the logic used below? I think should be move to ir.config.param

Reply:
@simahawk It's not global. It is specific to this scheduled action.

Having it here also makes it easier for the admin to configure everything in one place.

Reply:
@simahawk thanks for the thought. In addition to @max3903 if we move it to ir.config.param it has provided only 2 fields - key/value and we need value and selections(days/weeks/months) for users to set it.

Reply:
regarding 

> Having it here also makes it easier for the admin to configure everything in one place.

you can add the selection fields to res.config.settings and make stored on config param keys.[ This is done already in many places](https://github.com/odoo/odoo/blob/11.0/addons/auth_signup/models/res_config_settings.py#L35).

Reply:
another option, if you really like to config this from the cron, would be to pass them as keyword args to the cron call. Like:
```xml
<field name="code">model.send_notify_product_eol(approaching_type='days', approaching_number=1)</field>
```
so the user can still set everything from the cron. But, again, I think that this settings are one-shot setting and you don't play w/ them every day :)